### ISSUE TITLE
problem with makefile

### ISSUE THREAD

**Issue by tillmo** [spechub/Hets#1842]
A `make` leads to
```
[...]
/bin/ksh93: [: argument expected
```

**Comment by jelmd** [comment]
Usually happens when a var in a legacy test expression is unset, e.g. `[ -x $foo ]'. Is this reproducible?

**Comment by tillmo** [comment]
yes, this is reproducible, with `make distclean` and then `make`. Since the error message is issued by `ksh93`, I ask you, because you are the only one among us using that shell.

**Comment by jelmd** [comment]
This would happen on any shell. This is _not_ ksh specific! Anyway, need the output to get a clue.

Right now on my build systems build of the current master fails with:
```
Common/XmlExpat.hs:18:18:
    Could not find module `Text.XML.Expat.Tree'
    Use -v to see a list of the files searched for.

GUI/UDGUtils.hs:17:8:
    Could not find module `UDrawGraph.Graph'
    Use -v to see a list of the files searched for.

GUI/UDGUtils.hs:18:8:
    Could not find module `UDrawGraph.Basic'
    Use -v to see a list of the files searched for.

HolLight/HolLight2DG.hs:37:8:
    Could not find module `Text.XML.Expat.SAX'
    Perhaps you meant
      Text.XML.HaXml.SAX (from HaXml-1.25.3@HaXml_1GbZ2boGpXLD14xEQhqWVU)
    Use -v to see a list of the files searched for.

Taxonomy/AbstractGraphView.hs:58:18:
    Could not find module `UDrawGraph.Types'
    Use -v to see a list of the files searched for.
make[1]: *** [hets-opt] Error 1
make: *** [hets.bin] Error 2
```
If they are really needed, var.mk should check for availability ...

**Comment by tillmo** [comment]
You need to execute `make stack`. Note that the output from `ksh93` appears at the very beginning. It does not lead to an abortion of the compilation.

**Comment by jelmd** [comment]
Ahh, ok - that's a stack issue. IIRC stack does strange things and later runs make by itself without passing the proper environment (I guess, here it is the SHELL var, which doesn't get passed).  Without stack there is no such problem.

**Comment by tillmo** [comment]
The build failure that you have reported should vanish with `make stack`. 
However, the problem reported in this issue is unrelated to stack.

**Comment by jelmd** [comment]
If you feel more comfortable with bash, just force it by setting it hard at the 3rd line of var.mk.
